text/XXXX-externalize_private_keys.md

```diff
@@ -0,0 +1,111 @@
+
+### Phasing out `config.Profile.PrivKey`
+This work should happen in phases:
+
+1. **copy to keyring, support env var**. In in the next release, add in the new APIs, prefer their use when present.
+2. **intiailize with an empty privKey value**. Once we know the keyring approach works across a number of different platforms in a wide release, switch default initialization behaviour to _not_ write the privKey value.
+3. **remove the field** Eventually, remove the field entirely with a repo migration.
+
+In all phases the privKey field should be the _lowest_ in the loading hierarchy. Keyring values will override `Profile.PrivKey`, and the env var will override all other values.
+
+### Recovering from a deleted `.qri` folder
+Once we've externalized the private keys, we can change the behaviour of `qri setup` to first scan the keyring under the `io.qri.keys` namespace for existing keys. If a key is found, we can ask the user if they'd like to create a new account, or recover an old one.
+
+We can bolster this with calls to the default registry for the discovered public keys to add username info to the list. Profile information may not exist on the default registry, so we'll need to default to showing just the raw ProfileID. We can support this process with docs.
+
+Instead of jumping straight to building the UI for switching, we can supply a `--recover` flag that lets a user supply a ProfileID to recover from:
+
+```
+$ qri setup --recover QmProfileID
+```
+
+listing available private keys in a keyring may also form the basis of profile switching in the future.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+### More OS-Specific Work.
+This increases our reliance on an OS-level API that sometimes doesn't exist, and runs the risk of deviating behaviour between platforms. 
+
+Fallback to a `QRI_PRIVATE_KEY` file whenever no keyring exists is necessary to keep things working when our keyring implementation fails, effectivly reverting us back to the setup we're in today. This time we at least have a separate file and a warning.
+
+### Upstream work to get recovery functionality
+The package I've suggested we build on has no high-level method for listing the keys in a namespace. We'e need to be able to do this, and haven't confirmed this will be possible across all platforms.
+
+### No Qri-specific security
+This RFC does not propose adding any key-specific security. If we used symmetric encryption on the private key itself, Qri could require a local password of it's own. In this case we're relying on the user's Operating System security architecture
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+A number of alternatives exist for externalizing secrets. We could store keys in the `.ssh` directory if it exists, and looking for guidance on 
```

Comment:
I kind of went on a rumble with `.ssh` above without noticing this first. Also a point taken on the risk associated with piggybacking on `.ssh` as we have no control over it. I'm so accustomed to it that I considered it a default on any OS, which it definitely isn't. 

Back to square 1 I guess. It's just that I'd want the key in a private but _expected/known_ location.